packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -459,7 +465,18 @@ export const Table = <
           onSort?.(newState);
           break;
         case TableActions.setFilter:
-          onFilterHandler(newState, action, previousState, instance, onFilter);
+          // Check to see if filters have changed
```

Comment:
I think the same is needed for global filters but maybe with a separate PR because they are not even triggering `onFilter`

Reply:
There is an issue filed for that here #915 that will be handled in a separate PR